### PR TITLE
Add Dockerfile, which builds docker with nix package manager

### PR DESCRIPTION
**Pull request by offlinehacker**
With this Dockerfile, https://registry.hub.docker.com/u/offlinehacker/nixpkgs will automaticlly build docker image with nix package manager and nixpkgs(repo itself) on every push. To use this docker image: docker -t -i offlinehacker/nixpkgs. I advise nixos organization to create it's own account on docker registry and setup automated builds.

Example Dockefile that uses this image:

```
FROM offlinehacker/nixpkgs
RUN nix-env -iA nixpkgs.mongodb
RUN mkdir -p /var/db/mongodb
CMD mongod --dbpath /var/db/mongodb
```

PS: Docker is not a replacement for nixos, docker is process manager(which can also run whole nixos container), nixos is os, just to make this clear.

### CODE REVIEW

**Comment by 7c6f434c** [comment]
@offlinehacker So, are there any outstanding changes here?

I guess minor changes should be committed directly, and a PR without any changes should be closed. An issue could be created in its place, if needed.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM busybox
+
+RUN dir=`mktemp -d` && trap 'rm -rf "$dir"' EXIT && \
+    wget -O- http://nixos.org/releases/nix/nix-1.7/nix-1.7-x86_64-linux.tar.bz2  | bzcat | tar x -C $dir && \
```
Comment (wmertens) [inline Dockerfile]: This needs to be updated for Nix 1.8. Also, why not simply use the install procedure `curl https://nixos.org/nix/install | sh`?

Comment (offlinehacker) [inline Dockerfile]: I don't think that's even proper way to run nix with docker, I'm working on
service abstraction layer that would make declarative docker containers
possible.

**Comment by bbarker** [comment]
Using a very simple Dockerfile, I get an error that seems to indicate `/bin/sh` does not exist.


```
FROM offlinehacker/nixos
COPY *.nix /etc/nixos/
RUN "nixos-rebuild switch --upgrade"
```

```
$ docker build -t nixos:testing0 -f NixOS .                                                                                                                                                                                                                                                                                                                                                                       
Sending build context to Docker daemon  14.34kB
Step 1/3 : FROM offlinehacker/nixos
 ---> 65f8b74f4779
Step 2/3 : COPY *.nix /etc/nixos/
 ---> 4543000ca04b
Removing intermediate container de86ee1dd67c
Step 3/3 : RUN "nixos-rebuild switch --upgrade"
 ---> Running in 671ac3ef975d
container_linux.go:262: starting container process caused "exec: \"/bin/sh\": stat /bin/sh: no such file or directory"
oci runtime error: container_linux.go:262: starting container process caused "exec: \"/bin/sh\": stat /bin/sh: no such file or directory"
```

**Pull request closed without merging**